executor/builder.go

```diff
@@ -1236,15 +1236,79 @@ func (b *executorBuilder) buildUpdate(v *plan.Update) Executor {
 		b.err = errors.Trace(b.err)
 		return nil
 	}
+	columns2Handle := buildColumns2Handle(v.Schema(), tblID2table)
 	updateExec := &UpdateExec{
-		baseExecutor: newBaseExecutor(b.ctx, nil, v.ExplainID(), selExec),
-		SelectExec:   selExec,
-		OrderedList:  v.OrderedList,
-		tblID2table:  tblID2table,
+		baseExecutor:   newBaseExecutor(b.ctx, nil, v.ExplainID(), selExec),
+		SelectExec:     selExec,
+		OrderedList:    v.OrderedList,
+		tblID2table:    tblID2table,
+		columns2Handle: columns2Handle,
 	}
 	return updateExec
 }
 
+// Columns2HandleEntry represents an mapper from column index to handle index.
+type Columns2HandleEntry struct {
+	start, end int
+	handleIdx  int
```

Comment:
How about:
- s/handleIndex/handleOrdinal/
- s/Columns2HandleEntry/cols2Handle/
- s/Columns2Handle/cols2HandleSlice/


We can use the word "ordinal" to replace "index" to avoid misunderstanding.